src/views/Categories.vue

```diff
@@ -42,20 +67,57 @@ export default {
         this.categories = defaultOptions;
       }
     },
+    getQuestions() {
+      const defaultQuestions = {
+      };
+
+      this.categories.forEach((category) => {
+        defaultQuestions[category.value] = [];
+      });
+      console.log(defaultQuestions);
+
+      if (localStorage.getItem('questions')) {
+        this.questions = JSON.parse(localStorage.getItem('questions'));
+      } else {
+        this.questions = defaultQuestions;
+      }
+    },
     addCategory() {
       this.categories.push({
         text: this.newCategoryName,
         value: this.newCategoryName.trim().toLowerCase(),
       });
-
+      this.questions[this.newCategoryName.trim().toLowerCase()] = [];
+      localStorage.setItem('questions', JSON.stringify(this.questions));
       localStorage.setItem('categories', JSON.stringify(this.categories));
       this.newCategoryName = '';
     },
+    deleteCategoryControl() {
+      console.log(this.newCategoryName);
+      console.log(this.categories.length);
+      let i = 0;
+      this.questionsControl = JSON.parse(localStorage.getItem('questions'));
+      while (this.categories.length > i) {
+        if (this.newCategoryName.trim().toLowerCase() === this.categories[i].value) {
+          i = 0;
+          if (this.questionsControl[this.newCategoryName.trim().toLowerCase()].length === 0) {
+            this.checkCategory = true;
+          } else {
+            this.checkCategory = false;
+          }
+          return;
+        }
+        i += 1;
+      }
+    },
   },
   data() {
     return {
       newCategoryName: '',
       categories: [],
+      questionsControl: {},
+      questions: {},
+      checkCategory: true,
```

Comment:
`checkCategory` is not a great name. It is not obvious what it implies. A better name would be something like `canCategoryBeDeleted`.
Also data is not a great place for it. Either a method like `canBeleted(categoryName)` or a computed variable like `canCategoryBeDeleted` that uses `selectedCategory/currentCategory` from data is more appropriate